core/util.go

```diff
@@ -0,0 +1,8 @@
+package autospotting
+
+func minInt(a, b int) int {
```

Comment:
I think we already had such a function, please either use it or migrate the code to this one.